client/verta/verta/registry/entities/_modelversion.py

```diff
@@ -1514,3 +1516,98 @@ def set_hide_output_label(self, hide):
     def get_hide_output_label(self):
         self._refresh_cache()
         return self._msg.hide_output_label
+
+    def log_dataset(self, key, path, path_only, artifact_type, linked_artifact_id):
+        """
+        Logs a dataset to this Model Version.
+
+        Parameters
+        ----------
+        key : str
+            Artifact key.
+        path : str
+            Artifact path.
+        path_only : bool
+            Whether path only.
+        artifact_type : CommonService_pb2.ArtifactTypeEnum variant
+            Category to which the artifact belongs (e.g. MODEL, DATA).
+        linked_artifact_id : str
+            Linked artifact id.
+
+        """
+        if not key:
+            raise ValueError("key is not specified")
+        if not path:
+            raise ValueError("artifact_path is not specified")
+        if not path_only:
+            raise ValueError("path_only is not specified")
+        if not artifact_type:
+            raise ValueError("artifact_type is not specified")
+        if not linked_artifact_id:
+            raise ValueError("linked_artifact_id is not specified")
+
+        artifact_msg = _CommonCommonService.Artifact(
+            key=key,
+            path=path,
+            path_only=path_only,
+            artifact_type=artifact_type,
+            linked_artifact_id=linked_artifact_id,
+        )
+
+        msg = _RegistryService.LogDatasetsInModelVersion(
+            model_version_id=self.id,
+            datasets=[artifact_msg],
+        )
+
+        endpoint = "/api/v1/registry/model_versions/{}/logDatasets".format(
+            self.id,
+        )
+
+        response = self._conn.make_proto_request("POST", endpoint, body=msg)
+        self._conn.must_response(response)
+        self._clear_cache()
+
+    def get_dataset_keys(self):
+        """
+        Gets the dataset keys of this Model Version.
+
+        Returns
+        -------
+        list of str
+            List of dataset keys of this Model Version.
+
+        """
+        self._refresh_cache()
+        return list(map(lambda dataset: dataset.key, self._msg.datasets))
+
+    def get_dataset(self, key):
+        """
+        Get a dataset.
+
+        Parameters
+        ----------
+        key : str
+            Artifact key.
+
+        Returns
+        -------
+        `CommonService_pb2.Artifact`.
```

Comment:
We should never return an object from `*_pb2` to the user because they are not user-friendly. Please see Technical Implementation Plan item 3 in [the Jira ticket](https://vertaai.atlassian.net/browse/CAT-155) for how to return a client `DatasetVersion` object, if possible.